src/pamiq_core/model/interface.py

```diff
@@ -86,3 +89,31 @@ def forward(self, *args, **kwds) -> Any:
     def __call__(self, *args, **kwds) -> Any:
         """Calls `forward` method."""
         return self.forward(*args, **kwds)
+
+    def sync(self) -> None:
+        """Copies params of training model to self._inference_model if
+        needed."""
+        if self._need_sync:
+            self.sync_model(self.inference_model)
+
+    @property
+    def _need_sync(self) -> bool:
+        """Return whether It is necessary to synchronize training model and
+        inference model."""
+        return self.has_inference_model and (not self.inference_only)
+
+    def sync_model(self, inference_model: T) -> None:
+        """Copies params of training model to self._inference_model if needed.
+
+        Example:
+            class TrainingTorchModel(TrainingModel[InferenceTorchModel]):
+                @override
+                def sync_model(self, inference_model: InferenceTorchModel) -> None:
+                    with torch.no_grad():
+                        for inference_param, training_param in zip(
+                            inference_model.model.parameters(),
+                            self.model.parameters(),
+                        ):
+                            inference_param.copy_(training_param)
+        """
+        raise NotImplementedError
```

Comment:
メソッド名はこのようにすると良いと思う。
```suggestion
    @abstractmethod
    def sync_impl(self, inference_model: T) -> None:
        """Implementation of synchronization.
        
        Args:
            T: InferenceModel to sync.
        """
        ...
```

Reply:
https://github.com/MLShukai/pamiq-core/pull/72/commits/45441630b4a75bfe840c71be4af08aaf3905fbd4 で修正